integration-tests/gradle/src/integrationTest/kotlin/org/jetbrains/dokka/it/StdLibDocumentationIntegrationTest.kt

```diff
@@ -0,0 +1,38 @@
+package org.jetbrains.dokka.it
+
+import java.net.URL
+import kotlin.test.Test
+
+class StdLibDocumentationIntegrationTest {
+
+    /**
+     * Documentation for Enum's synthetic values() and valueOf() functions is only present in source code,
+     * but not present in the descriptors. However, Dokka needs to generate documentation for these functions,
+     * so it ships with hardcoded kdoc templates.
+     *
+     * This test exists to make sure documentation for these hardcoded synthetic functions does not change,
+     * and fails if it does, indicating that it needs to be updated.
+     */
+    @Test
+    fun shouldAssertEnumDocumentationHasNotChanged() {
+        val sourcesLink = "https://raw.githubusercontent.com/JetBrains/kotlin/master/core/builtins/native/kotlin/Enum.kt"
+        val sources = URL(sourcesLink).readText()
+
+        val expectedValuesDoc =
+            "    /**\n" +
```

Comment:
For instance, I can change the templates accidentally and the test can not catch it. There are other ways to break this test.

Reply:
> I can change the templates accidentally and the test can not catch it

This test is not supposed to catch template changes, it checks that Enum's KDocs from stdlib haven't changed in `master` branch, in case any information is added or removed. It doesn't have to be word-for-word as the template we're using, we just need to know when it changes so that we can adjust the template accordingly (if needed).

> There are other ways to break this test.

No doubt, there are many ways to break any test, but I honestly don't see how creating another file and putting the string in there will help :) It will, however, complicate the test a little, thus making it a bit more difficult to understand at a glance

___

Maybe I misunderstood you and you are proposing to reuse the same file from which documentation is generated? i.e `resources/dokka/docs/kdoc/EnumValueOf.kt.template`? In this case I'm not sure how I can have access to it... But even if we do somehow get it, it's not a direct copy-paste, so the assertion will fail.

Reply:
>  to reuse the same file from which documentation is generated
Indeed.

In my opinion, there are two ways to fix the test:
1. Extract the text from generated HTML/Documentable. 
2. Use `resources/dokka/docs/kdoc/EnumValueOf.kt.template`. You can get access to it through [sourceSet configuration in Gradle script](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/SourceSet.html#getResources--).